tests/_save/loaders/test_json_loader.py

```diff
@@ -64,142 +65,146 @@ def test_cache_hit_miss(self) -> None:
-        with open(missing_fields_path, "w") as f:
-            json.dump({"hash": "missing", "stateful_refs": []}, f)
-
-        with pytest.raises(LoaderError, match="Invalid json object"):
-            loader.load_persistent_cache("missing", "Pure")
-
-    def test_load_cache(self) -> None:
-        """Test the load_cache method."""
-        loader = JsonLoader("test", self.save_path)
-
-        # Create a cache file
-        cache_path = loader.build_path("hash1", "Pure")
-
-        # Create a valid JSON cache
-        cache_dict = {
-            "defs": {"var1": "value1"},
-            "hash": "hash1",
-            "stateful_refs": [],
-            "cache_type": "Pure",
-            "hit": True,
-            "meta": {},
-        }
-
-        with open(cache_path, "w") as f:
-            json.dump(cache_dict, f)
-
-        # Load the cache
-        loaded_cache = loader.load_cache("hash1", "Pure")
-        assert loaded_cache.hash == "hash1"
-
-        # Should raise for non-existent cache
-        with pytest.raises(LoaderError, match="Unexpected cache miss"):
-            loader.load_cache("nonexistent", "Pure")
-
-    def test_save_cache(self) -> None:
-        """Test saving a cache."""
-        loader = JsonLoader("test", self.save_path)
-
-        # Create a cache with a stateful reference
-        cache = Cache(
-            {"var1": "value1"},
-            "hash1",
-            {"stateful"},
-            "Pure",
-            True,
-            {"version": 1},
-        )
-
-        # Save the cache
-        loader.save_cache(cache)
-
-        # Verify the file was created
-        cache_path = loader.build_path("hash1", "Pure")
-        assert os.path.exists(cache_path)
-
-        # Load the JSON and verify contents
-        with open(cache_path) as f:
-            loaded_json = json.load(f)
-
-        assert loaded_json["hash"] == "hash1"
-        assert loaded_json["cache_type"] == "Pure"
-        assert loaded_json["hit"] is True
-        assert isinstance(
-            loaded_json["stateful_refs"], list
-        )  # Should be converted to list
-        assert loaded_json["meta"] == {"version": 1}
-
-        # Save another cache with different type
-        cache2 = Cache(
-            {"var2": "value2"}, "hash2", set(), "Deferred", True, {}
-        )
-
-        loader.save_cache(cache2)
-
-        # Verify the second file was created
-        cache2_path = loader.build_path("hash2", "Deferred")
-        assert os.path.exists(cache2_path)
-
-        # Load the second JSON and verify contents
-        with open(cache2_path) as f:
-            loaded_json2 = json.load(f)
-
-        assert loaded_json2["hash"] == "hash2"
-        assert loaded_json2["cache_type"] == "Deferred"
+        assert not loader.cache_hit(key("empty", "Pure"))
+
+
+# These are reasoanble, granular unit tests
+# But should be covered by the general testing
+#
+#    def test_load_persistent_cache(self) -> None:
```

Comment:
if these are covered and currently a maintenance burden because of the changes, feel free to remove them